crates/aptos/src/update/revela.rs

```diff
@@ -0,0 +1,173 @@
+const REVELA_EXE: &str = "revela.exe";
+#[cfg(not(target_os = "windows"))]
+const REVELA_EXE: &str = "revela";
+
+/// Update Revela, the tool used for decompilation.
+#[derive(Debug, Parser)]
+pub struct RevelaUpdateTool {
+    /// The owner of the repo to download the binary from.
+    #[clap(long, default_value = "verichains")]
+    repo_owner: String,
+
+    /// The name of the repo to download the binary from.
+    #[clap(long, default_value = "revela")]
+    repo_name: String,
+
+    /// The version to install, e.g. 1.0.1. Use with caution, the default value is a
+    /// version that is tested for compatibility with the version of the CLI you are
+    /// using.
+    #[clap(long, default_value = TARGET_REVELA_VERSION)]
+    target_version: String,
+
+    /// Where to install the binary. Make sure this directory is on your PATH. If not
+    /// given we will put it in a standard location for your OS that the CLI will use
+    /// later when the tool is required.
+    #[clap(long)]
+    install_dir: Option<PathBuf>,
+}
+
+impl BinaryUpdater for RevelaUpdateTool {
+    fn pretty_name(&self) -> &'static str {
+        "Revela"
+    }
+
+    /// Return information about whether an update is required.
+    fn get_update_info(&self) -> Result<UpdateRequiredInfo> {
+        // Get the current version, if any.
+        let revela_path = get_revela_path();
+        let current_version = match revela_path {
+            Ok(path) => {
+                let output = std::process::Command::new(path)
+                    .arg("--version")
+                    .output()
+                    .context("Failed to get current version of Revela")?;
+                let stdout = String::from_utf8(output.stdout)
+                    .context("Failed to parse current version of Revela as UTF-8")?;
+                let current_version = stdout
+                    .split_whitespace()
+                    .nth(1)
+                    .map(|s| s.to_string())
+                    .context("Failed to extract version number from command output")?;
+                Some(current_version.trim_start_matches('v').to_string())
+            },
+            Err(_) => None,
+        };
+
+        // Strip v prefix from target version if present.
+        let target_version = self.target_version.trim_start_matches('v').to_string();
+
+        Ok(UpdateRequiredInfo {
+            current_version,
+            target_version,
+        })
+    }
+
+    fn build_updater(&self, info: &UpdateRequiredInfo) -> Result<Box<dyn ReleaseUpdate>> {
+        // Determine the target we should download based on how the CLI itself was built.
+        let arch_str = get_arch();
+        let build_info = cli_build_information();
+        let target = match build_info.get(BUILD_OS).context("Failed to determine build info of current CLI")?.as_str() {
+            "linux-aarch64" | "linux-x86_64" => "unknown-linux-gnu",
+            "macos-aarch64" | "macos-x86" => "apple-darwin",
```

Comment:
Don't we need different binaries for Mx and X86 arch? Or is the dawin a multi-arch binary? I saw that revela has two binaries for the diferent Mac versions

Reply:
Using the two different binaries is an optimization, but Intel builds should still work on ARM for now.

Would be good if they could manage brew as well.

Reply:
We do actually, this part just figures out the OS, we determine the CPU architecture separately and grab the appropriate binary. So on ARM MacOS you'll get an ARM build, on x86 an x86 build.